### PR TITLE
Generate the ToDerivedType method override only for ancestors which do add a field.

### PR DESCRIPTION
**Pull request by azeno**
See issue #46

### CODE REVIEW

File: src/ImmutableObjectGraph.CodeGeneration.Tests/TestSources/Hierarchy3Levels.cs [hunk 1]
```diff
@@ -8,7 +8,19 @@ abstract partial class L2 : L1
...
+}
+
+[ImmutableObjectGraph.CodeGeneration.GenerateImmutable]
+partial class L32 : L2
+{
+    bool baseField;
+}
+
+[ImmutableObjectGraph.CodeGeneration.GenerateImmutable]
+partial class L4 : L32
```
Comment (AArnott) [inline src/ImmutableObjectGraph.CodeGeneration.Tests/TestSources/Hierarchy3Levels.cs]: Hmm... It looks like `Hierarchy3Levels.cs` is no longer an accurate name for this file any more. :) Can you update it?

**Comment by azeno** [comment]
Ähm not sure why there should be a conflict now?

**Comment by azeno** [comment]
Can you tell me what appveyor is complaining about so I can try to fix it?

**Comment by AArnott** [comment]
This isn't appveyor complaining. It's that your branch and master has diverged so that git can't automatically merge your changes into this library. Do you know how to resolve that merge conflict?

**Comment by azeno** [comment]
Argh. Forgot to merge back master. Working now.

**Comment by AArnott** [comment]
@azeno It looks like you failed to check in the change to the project file for the source file rename.

**Pull request merged**